dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/AIFunctionKernelFunction.cs

```diff
@@ -23,28 +25,15 @@ internal sealed class AIFunctionKernelFunction : KernelFunction
     private readonly AIFunction _aiFunction;
 
     public AIFunctionKernelFunction(AIFunction aiFunction) :
-        base(aiFunction.Metadata.Name,
-            aiFunction.Metadata.Description,
-            aiFunction.Metadata.Parameters.Select(p => new KernelParameterMetadata(p.Name, AbstractionsJsonContext.Default.Options)
-            {
-                Description = p.Description,
-                DefaultValue = p.DefaultValue,
-                IsRequired = p.IsRequired,
-                ParameterType = p.ParameterType,
-                Schema =
-                    p.Schema is JsonElement je ? new KernelJsonSchema(je) :
-                    p.Schema is string s ? new KernelJsonSchema(JsonSerializer.Deserialize(s, AbstractionsJsonContext.Default.JsonElement)) :
-                    null,
-            }).ToList(),
-            AbstractionsJsonContext.Default.Options,
+        base(aiFunction.Name,
+            aiFunction.Description,
+            [.. MapParameterMetadata(aiFunction)],
```

Comment:
Why not just have MapParameterMetadata return the correct type rather having the extra enumerable layer?